packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts

```diff
@@ -24,20 +25,25 @@ export function isValidCollectionPathImplementation(
 
     try {
       assert(opener, `{`, ``) // This is a noop because of the opening check, but here for posterity
-      assert(model, /^[A-Z][a-zA-Z]+$/, errorMessage(filePath, part))
-      assert(field, /^[a-zA-Z_()]+$/, errorMessage(filePath, part))
-      assert(closer, `}`, errorMessage(filePath, part))
+      assert(model, /^[A-Z][a-zA-Z]+$/, errorMessage(part))
+      assert(field, /^[a-zA-Z_()]+$/, errorMessage(part))
+      assert(closer, `}`, errorMessage(part))
     } catch (e) {
-      reporter.panicOnBuild(e.message)
+      reporter.panicOnBuild({
+        id: prefixId(CODES.CollectionPath),
+        context: {
+          sourceMessage: e.message,
+        },
+        filePath: filePath,
+      })
       passing = false
     }
   })
   return passing
 }
 
-function errorMessage(filePath: string, part: string): string {
-  return `PageCreator: Collection page builder encountered an error parsing the filepath. To use collection paths the schema to follow is {Model.field}. The problematic part is: ${part}.
-filePath: ${filePath}`
+function errorMessage(part: string): string {
+  return `Collection page builder encountered an error parsing the filepath. To use collection paths the schema to follow is {Model.field}. The problematic part is: ${part}.`
```

Comment:
Is there no value in tracking the paths, anyways, as they might surface problems related to particular cases? As like a second level unfiltered set of examples next to the aggregated grouped problem code.

Reply:
I've moved the filePath to out of the `context` to [here](https://github.com/gatsbyjs/gatsby/blob/unified-routes/structured-errors/packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts#L37) so that the [cli can take over](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-cli/src/reporter/loggers/ink/components/error.tsx#L67-L72).